app/views/users/_vitals.html.erb

```diff
@@ -4,21 +4,20 @@
 
 <div class="list-group-item">
   <span class="badge"><%= number_of_collections %></span>
-  <i class="glyphicon glyphicon-folder-open"></i> <%= t("sufia.dashboard.stats.collections") %>
+  <i class="glyphicon glyphicon-folder-open"></i> <%= link_to_field('depositor', user.to_s, t("sufia.dashboard.stats.collections"), 'f[human_readable_type_sim][]': "Collection") %>
 </div>
-  
+
 <div class="list-group-item">
-  <span class="badge"><%= number_of_files(user) %></span>
-    <i class="glyphicon glyphicon-upload"></i> <%= link_to_field('depositor', user.to_s, 'Deposited Files') %>
+  <span class="badge"><%= number_of_works %></span>
```

Comment:
I don't think you want to works of the current user. You want the number of works for the user you are looking at on this page.